custom_components/unfoldedcircle/config_flow.py

```diff
@@ -307,23 +308,48 @@ async def async_step_reauth_confirm(
         )
 
 
-class UnfoldedCircleRemoteOptionsFlowHandler(OptionsFlow):
+class UnfoldedCircleRemoteOptionsFlowHandler(config_entries.OptionsFlow):
     """Handle Unfolded Circle Remote options."""
 
-    def __init__(self, config_entry: ConfigEntry) -> None:
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
         """Initialize options flow."""
         self.config_entry = config_entry
+        self.options = dict(config_entry.options)
 
-    async def async_step_init(
-        self, user_input: dict[str, int] | None = None
-    ) -> FlowResult:
-        """Manage Unfolded Circle options."""
+    async def async_step_init(self, user_input=None):  # pylint: disable=unused-argument
+        """Manage the options."""
+        return await self.async_step_user()
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
         if user_input is not None:
-            return self.async_create_entry(title="", data=user_input)
+            self.options.update(user_input)
+            return await self._update_options()
 
         return self.async_show_form(
-            step_id="init",
-            data_schema=STEP_USER_DATA_SCHEMA,
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_GLOBAL_MEDIA_ENTITY,
+                        default=self.config_entry.options.get(
+                            CONF_GLOBAL_MEDIA_ENTITY, True
+                        ),
+                    ): bool,
+                    vol.Optional(
+                        CONF_ACTIVITY_GROUP_MEDIA_ENTITIES,
+                        default=self.config_entry.options.get(
+                            CONF_ACTIVITY_GROUP_MEDIA_ENTITIES, True
```

Comment:
This still leaves a gap for those that would like multiple media players, but don't have activity groups defined. (Or do have some defined but also have stand alone activities) If a single media player solved this case, there wouldn't be a need to create them for each activity group.

Reply:
But in that case the first option (enabled by default) will address it : we keep what I called the "global media player entity" which will address all the activities defined (wether they are in an activity group or standalone activities) and then option 2 to get additional media player entities for each activity group. If you don't have any activity group you should get one media player entity at least.

However improvements need to be done on the selection of the "right" media player entity within an activity. The rules that I defined are....hmmm could be better I think

Reply:
By the way I saw but I may have misunderstood that you are using the source list of the media player entity to switch from one media player to another. Am I correct ? This sounds like a good idea however it raises a regression : media players do have source list most of the time and you loose this ability to select the source of the media player. For apple TV like Shield TV it will be the app (Netflix, Prime...). For my STB it is the TV channel.
I don't have any idea to workaround this